doc/configuration.rst

```diff
@@ -18,12 +18,94 @@ The name can be seem from ``doit help`` output::
    -f ARG, --file=ARG        load task from dodo FILE [default: dodo.py]  (config: dodoFile)
 
 
+pyproject.toml
+--------------
+
+`doit` configuration can be read from `pyproject.toml <https://www.python.org/dev/peps/pep-0518/>`_
+under the `tool.doit` namespace. In the future, especially if a TOML parser becomes
+part of the python standard library, this will become the preferred configuration source,
+and may gain features not available in the legacy `doit.cfg`.
+
+.. note::
+
+   As a TOML parser is _not_ yet part of the standard library, a third-party package is
+   required, one of:
+
+   - `toml <https://pypi.org/project/toml/>`_
+   - `tomlkit <https://pypi.org/project/tomlkit/>`_
+
+
+TOML vs INI
+^^^^^^^^^^^
+
+While mostly similar, `TOML <https://toml.io>`_ differs from the INI format
+in a few ways:
+
+- all strings must be quoted with `'` or `"`
+- triple-quoted strings may contain new line characters (`\n`) and quotes
+- must be saved as UTF-8
+- integers and floating point numbers can be written without quotes
+- boolean values can be written unquoted and lower-cased, as `true` and `false`
+
+Unlike "plain" TOML, `doit` will parse pythonic strings into their correct types,
+e.g. `"True"`, `"False"`, `"3"`, but using "native" TOML types may be preferable.
+
+
+tool.doit
+^^^^^^^^^
+
+The `tool.doit` section may contain command line options that will be used
```

Comment:
I guess mentioned this before...

Why not just `doit` instead of `tool.doit`? For me it seems unnecessary verbose.

Reply:
This is specified by [PEP 518](https://www.python.org/dev/peps/pep-0518/#tool-table).

Reply:
I see it is their recommendation... although I dont really understand/agree why limit to only 2 top tables.

So another question. Why put `doit` configuration on `pyproject.toml` instead of going with out own file `doit.toml`?

Reply:
>  Why put doit configuration on pyproject.toml 

Indeed, that has a solid question, and mypy has fought back viciously, for some reason. But my feeling is that the more tools I can cram into a single, consensus-driven file, the better I'll be able to automate the creation (through schema, etc), updating, etc. vs code.

> doit.toml?

A fine idea as well, and could be added to the list. I'd imagine _this_ one _wouldn't_ have the `tool.doit` prefix.